test/unit/auth/credential.spec.ts

```diff
@@ -563,6 +572,20 @@ describe('Credential', () => {
       expect(isApplicationDefault(c)).to.be.true;
     });
 
+    it('should return false for RefreshTokenCredential loaded from FIREBASE_EMULATOR_CREDENTIALS', () => {
```

Comment:
I think this ought to return true (which means setting implicit=true in the ctor). Otherwise calls like `admin.firestore()` will fail with an invalid-credential error. See https://github.com/firebase/firebase-admin-node/blob/d98560260b66435255e928778c7b5845926204d7/src/firestore/firestore.ts#L66-L102

Also note that Firestore and Storage run their own credential discovery routine, which might fallback to gcloud credentials. Is that the behavior we want? Unfortunately those packages don't support passing explicit refresh token credentials.

Reply:
So I have tested this with the Firestore SDK and it works ... maybe that's only the case for Storage?  I thought all of those SDKs shared credential discovery.

Reply:
`@google-cloud/*` libraries have their own credentials implementation and lookup. We can pass a service account, but not a refresh token. So in this case they will silently fallback to using the local gcloud sdk credentials.